In [2]:
import pandas as pd
import sqlite3

connection = sqlite3.connect("./data/sql-murder-mystery.db")
cursor = connection.cursor()

In [19]:
query='''
SELECT date, city , type, description
  FROM crime_scene_report
  WHERE date = 20180115 and city ="SQL City"
  '''

cursor.execute(query)
cursor.fetchall()


#	Security footage shows that there were 2 witnesses. The first witness lives at the last house on "Northwestern Dr". The second witness, named Annabel, lives somewhere on "Franklin Ave".

[(20180115,
  'SQL City',
  'assault',
  'Hamilton: Lee, do you yield? Burr: You shot him in the side! Yes he yields!'),
 (20180115, 'SQL City', 'assault', 'Report Not Found'),
 (20180115,
  'SQL City',
  'murder',
  'Security footage shows that there were 2 witnesses. The first witness lives at the last house on "Northwestern Dr". The second witness, named Annabel, lives somewhere on "Franklin Ave".')]

In [8]:
query='''
SELECT name, id, address_street_name
  FROM person
  WHERE address_street_name = "Franklin Ave" and name like "Annabel%"
'''
cursor.fetchall()

#Id = 16371

[('Annabel Miller', 16371, 'Franklin Ave')]

In [10]:
cursor = connection.cursor()

In [18]:
query='''
SELECT transcript
  FROM interview
  WHERE person_id = 16371
'''

cursor.execute(query)
cursor.fetchall()
#I saw the murder happen, and I recognized the killer from my gym when I was working out last week on January the 9th.


[('I saw the murder happen, and I recognized the killer from my gym when I was working out last week on January the 9th.',)]

In [17]:
query='''
SELECT membership_id
  FROM get_fit_now_check_in
  WHERE check_in_date like "20180109"
'''

cursor.execute(query)
cursor.fetchall()


[('X0643',),
 ('UK1F2',),
 ('XTE42',),
 ('1AE2H',),
 ('6LSTG',),
 ('7MWHJ',),
 ('GE5Q8',),
 ('48Z7A',),
 ('48Z55',),
 ('90081',)]

membership_id
X0643
UK1F2
XTE42
1AE2H
6LSTG
7MWHJ
GE5Q8
48Z7A
48Z55
90081

The first witness lives at the last house on "Northwestern Dr".

In [22]:

query= '''
SELECT MAX (address_number), id
  FROM person
  WHERE address_street_name = "Northwestern Dr"
  order by address_number
'''
cursor.execute(query)  
cursor.fetchall()



[(4919, 14887)]

address_number	id

4919	14887

In [24]:
query = '''
SELECT transcript
  FROM interview
  WHERE person_id = 14887
'''
cursor.execute(query)
cursor.fetchall()



[('I heard a gunshot and then saw a man run out. He had a "Get Fit Now Gym" bag. The membership number on the bag started with "48Z". Only gold members have those bags. The man got into a car with a plate that included "H42W".',)]

I heard a gunshot and then saw a man run out. He had a "Get Fit Now Gym" bag. The membership number on the bag started with "48Z". Only gold members have those bags. The man got into a car with a plate that included "H42W".

In [25]:
query= '''SELECT *
  FROM get_fit_now_check_in
  WHERE membership_id like"48Z%" and check_in_date like "20180109"
'''

cursor.execute(query)
cursor.fetchall()

[('48Z7A', 20180109, 1600, 1730), ('48Z55', 20180109, 1530, 1700)]

membership_id	check_in_date	check_in_time	check_out_time
48Z7A	20180109	1600	1730
48Z55	20180109	1530	1700

In [26]:
query= '''
SELECT *
  FROM get_fit_now_member
  WHERE id = "48Z7A" or id = "48Z55"
'''

cursor.execute(query)
cursor.fetchall()

[('48Z55', 67318, 'Jeremy Bowers', 20160101, 'gold'),
 ('48Z7A', 28819, 'Joe Germuska', 20160305, 'gold')]

id	person_id	name	membership_start_date	membership_status
48Z55	67318	Jeremy Bowers	20160101	gold
48Z7A	28819	Joe Germuska	20160305	gold

In [27]:
query= '''SELECT id
  FROM drivers_license
  WHERE plate_number like "%H42W%" 
'''
cursor.execute(query)
cursor.fetchall()




[(183779,), (423327,), (664760,)]

id
183779
423327
664760

In [31]:
query= '''SELECT *
  FROM person
  WHERE license_id IN ("183779", "423327", "664760") AND
  id IN ("67318", "28819") AND
  name IN ("Jeremy Bowers", "Joe Germuska");
'''
cursor.execute(query)
print("Y aqui tenemos todos los datos del assesino:")
cursor.fetchall()






Y aqui tenemos todos los datos del assesino:


[(67318, 'Jeremy Bowers', 423327, 530, 'Washington Pl, Apt 3A', 871539279)]

id	name	license_id	address_number	address_street_name	ssn
51739	Tushar Chandra	664760	312	Phi St	137882671
67318	Jeremy Bowers	423327	530	Washington Pl, Apt 3A	871539279
78193	Maxine Whitely	183779	110	Fisk Rd	137882671
